testing/trino-product-tests/src/main/java/io/trino/tests/product/clickhouse/TestInsert.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.clickhouse;
+
+import com.google.common.net.InetAddresses;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.query.QueryResult;
+import io.trino.tempto.query.QueryResult.QueryResultBuilder;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.sql.JDBCType;
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryResult.fromSqlIndex;
+import static io.trino.tests.product.TestGroups.CLICKHOUSE;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.clickhouse.ClickHouseDataTypesTableDefinition.ALL_TYPES_DDL;
+import static io.trino.tests.product.clickhouse.ClickHouseDataTypesTableDefinition.ALL_TYPES_LOCAL_DDL;
+import static io.trino.tests.product.clickhouse.TestClickHouseTable.distributedClickHouseTable;
+import static io.trino.tests.product.clickhouse.TestClickHouseTable.localClickHouseTable;
+import static io.trino.tests.product.clickhouse.TestConstants.CLUSTER_NAME;
+import static io.trino.tests.product.clickhouse.TestConstants.CONNECTOR_NAME;
+import static io.trino.tests.product.clickhouse.TestConstants.SCHEMA_NAME;
+import static io.trino.tests.product.utils.QueryExecutors.onClickHouse;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+
+public class TestInsert
+        extends ProductTest
+{
+    @Test(groups = {CLICKHOUSE, PROFILE_SPECIFIC_TESTS})
+    public void testInsertMin()
```

Comment:
Let's start with less code as `TestPhoenix`. Basically, these test cases are already covered in `TestClickHouseTypeMapping`.

Reply:
Does this mean that only `CREATE TABLE AS SELECT` needs to be tested in `TestInsert`?

Reply:
@tangjiangling I meant you can put only `TestClickHouse.testCreateTableAsSelect()` instead of adding these tests.

Reply:
> Let's start with less code as TestPhoenix. Basically, these test cases are already covered in TestClickHouseTypeMapping.

Although we have done rich unit tests for single node deployment approach. But as a supplement, I think it is still necessary to cover some simple read/write tests for the cluster deployment approach in the product testing.

As you mentioned before #7600, this is a user feedback issue and I think another goal of product testing is to find problems before the user does, so I think these tests are necessary.

@hashhar What is your opinion?

Reply:
Surely, I don't disagree on adding **_simple_** read/writes test.

Reply:
@ebyhr Well, I will update this PR as you said.

Reply:
@ebyhr 😭 Sorry, I think I misunderstood you before. We don't support distributed writes yet, so it doesn't make sense to add these tests.

I've put the previous code in [this branch](https://github.com/tangjiangling/trino/tree/add-product-tests-for-clickhouse-clusters-backup) of my personal repository, feel free to ping me if distributed writes are supported, I'm glad to continue the previous work.